### ISSUE TITLE
Trusty as default Linux is coming leaves no way to run PHP 5.2 jobs

### ISSUE THREAD

**Issue by ntwb** [travis-ci/travis-ci#8072]
It was announced overnight that Travis CI are going to switch to Trusty as the default:
> https://blog.travis-ci.com/2017-07-11-trusty-as-default-linux-is-coming

To not be forced to the new Trusty distro I updated my `.travis.yml` configuration to this:
```yml
sudo: false
dist: trusty

language: php

matrix:
  include:
    - php: 7.1
    - php: 5.6
    - php: 5.2
      sudo: required
      dist: precise

script: phpunit
```

Now unfortunately this leaves those who test against PHP 5.2 in a a bit of a pickle:

PHP 5.2 current availability:

• ✅ `dist: precise` with `sudo: false`
• ❌  `dist: trusty` with `sudo: false`
• ❌  `dist: precise` with `sudo: required`

The only workaround I see currently is not specify `sudo: required` for the PHP 5.2 job, though this will be overwritten beginning September 2017 if all goes to plan pe the announcment:
> The tentative plan is to begin overwriting configurations that specify `dist: precise` to include `sudo: required` beginning in September 2017.

Is there any other solution to make available PHP 5.2 in a Travis CI job?

**Comment by johnbillion** [comment]
It's worth noting that the [WordPress open source project](https://github.com/WordPress/wordpress-develop) still supports PHP 5.2 and is affected by this.

**Comment by BanzaiMan** [comment]
Thanks for taking the time to open this issue.

While we sympathize with the predicament you are in, we are not able to provide PHP 5.2/5.3 runtime on Trusty. The technical details are given in https://github.com/travis-ci/travis-ci/issues/2963.

If you really need it, you can try pulling our Docker image (quay.io/travisci/travis-php) to run the jobs. Be forewarned, however, that the image is rather large and it could add significant time to your builds.

**Comment by ntwb** [comment]
Thanks Hiro,will take a look 👍

**Comment by nazar-pc** [comment]
PHP 5.3 is not supported at all since 14 August 2014, 5.2 since 2011. I'd just close this issue immediately. There is no reason to support these versions anymore, especially considering it is not straightforward to get them on Trusty.

I understand that people have legacy stuff and need time to migrate, but 5.2, in 2017, seriously?

The fact that WordPress supports some old completely unsupported and insecure version of PHP doesn't mean anyone should be advised to use it.

**Comment by jrfnl** [comment]
> The fact that WordPress supports some old completely unsupported and insecure version of PHP doesn't mean anyone should be advised to use it.

There is a big difference between _advising people to use something_ and _supporting_ something.

WordPress actively encourages people to use supported PHP versions (PHP 5.6 and higher), but still _supports_ the software running on older PHP versions (and prefers to make sure it does so correctly by build testing on all PHP versions it supports).

**Comment by ntwb** [comment]
For reference if anyone else is looking for a way to run PHP 5.2 or 5.2 on Travis CI Trusty builds

We (WordPress) have started working on a solution that will allow use PHP 5.2 and 5.3 on Trusty.

The first iteration of this is in a plugin currently undergoing heavy development, the details can be viewed in this commit https://github.com/WordPress/gutenberg/commit/3c3427472fadd4696347956b3b3e409bd21d606d. 

tl;dr: It uses `phpbrew` to install and switch to PHP 5.2 or 5.2, and then conditionally install the required `phpunit` version with an `openssl` fix on the side
